### PR TITLE
0825 cc remove legacy sc tests

### LINKED ISSUE #825: Clean up legacy smart contract tests and bytecodes
**Summary**
Remaining work for #824

**Possible resolution**
- [x] Delete all legacy smart contract tests
- [x] Delete (or move to `test-clients/src/main/resource/contract/bytecodes`) all bytecodes in the following folders that are being used by the legacy smart contract tests
- - [x] `test-clients/testfiles`
- - [x] `test-clients/src/main/resource`
- - [x] `test-clients/src/main/resource/testfiles`

### PR DESCRIPTION
**Pull request by anighanta**
**Related issue(s)**:
Closes #825 

**Summary of the change**:
1. Removed all legacy Smart Contract tests except for `SmartContractTestBitcarbon` which is still to be converted into an eet suite.
2. Moved all byte code to `test-clients/src/main/resource/contract/bytecodes` and updated the paths to all the byte code in the tests.

**External impacts**:
None.

**Applicable documentation**
- [ ] Javadoc
- [ ] HAPI protobuf comments
- [ ] README

### CODE REVIEW

**Review by qnswirlds — CHANGES_REQUESTED** [review]
Please merge master to this branch and resolve conflicts. The first clean up round for smart contract just got merged last night. Also, the reason we have this separate issue for the remaining work of smart contract clean up is because I wasn't sure that each of the legacy smart contract tests already had an EET replacement or not. Can you please confirm that?

**Comment by anighanta** [comment]
> Please merge master to this branch and resolve conflicts. The first clean up round for smart contract just got merged last night. Also, the reason we have this separate issue for the remaining work of smart contract clean up is because I wasn't sure that each of the legacy smart contract tests already had an EET replacement or not. Can you please confirm that?

Thanks @qnswirlds, I've merged master. All of the SmartContract tests that we use for regular regression testing purposes have been converted to an equivalent EET replacement. Remaining Legacy SmartContract tests are not being used. But with the available solidity files I think we can always create a new EET test case when we want to explore a new scenario. I Discussed with @ljianghedera if we can delete these tests.

**Comment by qnswirlds** [comment]
> > Please merge master to this branch and resolve conflicts. The first clean up round for smart contract just got merged last night. Also, the reason we have this separate issue for the remaining work of smart contract clean up is because I wasn't sure that each of the legacy smart contract tests already had an EET replacement or not. Can you please confirm that?
> 
> Thanks @qnswirlds, I've merged master. All of the SmartContract tests that we use for regular regression testing purposes have been converted to an equivalent EET replacement. Remaining Legacy SmartContract tests are not being used. But with the available solidity files I think we can always create a new EET test case when we want to explore a new scenario. I Discussed with @ljianghedera if we can delete these tests.

I just reviewed again. There are still some legacy tests hanging around, as follows:,
```
test-clients/src/main/java/com/hedera/services/legacy/autorenew/SmartContractAutoRenewCheck.java
test-clients/src/main/java/com/hedera/services/legacy/client/test/FileDeleteBalanceCheck.java
test-clients/src/main/java/com/hedera/services/legacy/file/LargeFileUploadIT.java
test-clients/src/main/java/com/hedera/services/legacy/file/SystemDeleteIT.java
test-clients/src/main/java/com/hedera/services/legacy/regression/CryptoTransferAll.java
test-clients/src/main/java/com/hedera/services/legacy/regression/DynamicRestartTest.java
test-clients/src/main/java/com/hedera/services/legacy/regression/SmartContractAggregatedTests.java
test-clients/src/main/java/com/hedera/services/legacy/regression/SmartContractTestBitcarbon.java
test-clients/src/main/java/com/hedera/services/legacy/regression/TransactionInsufficientFeeCost.java
test-clients/src/main/java/com/hedera/services/legacy/regression/umbrella/SmartContractServiceTest.java
```
Are they still in effect? If not, we should delete more. For those that are still being used, please move the bytecode paths and ABI's to ContractResources as we did for EET.  Most of the bytecode paths and ABI's should be there already, so just use them instead of adding more. Thanks!

**Comment by anighanta** [comment]
test-clients/src/main/java/com/hedera/services/legacy/regression/SmartContractAggregatedTests.java
test-clients/src/main/java/com/hedera/services/legacy/regression/SmartContractTestBitcarbon.java
test-clients/src/main/java/com/hedera/services/legacy/file/LargeFileUploadIT.java
These are in use @qnswirlds 


test-clients/src/main/java/com/hedera/services/legacy/regression/umbrella/SmartContractServiceTest.java
is very tightly nested with legacy umbrella tests. We can remove this when we refactor those legacy tests.

File: test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/meta/ContractResources.java [hunk 1]
```diff
@@ -205,6 +209,122 @@
...
+	public static final String MINT_CONFIGURE_ABI = "{\"constant\":false,\"inputs\":[" +
+			"{\"name\":\"_jurisdictions\",\"type\":\"address\"}]," +
+			"\"name\":\"configureJurisdictionContract\",\"outputs\":[]," +
+			"\"payable\":false,\"stateMutability\":\"nonpayable\",\"type\":\"function\"}";
+	public static final String PAY_TEST_GET_BALANCE = "{\"constant\":true,\"inputs\":[],\"name\":\"getBalance\"," +
+			"\"outputs\":[{\"name\":\"\",\"type\":\"uint256\"}]," +
+			"\"payable\":false,\"stateMutability\":\"view\",\"type\":\"function\"}";
+	public static final String PAY_TEST_DEPOSIT = "{\"constant\":false,\"inputs\":[" +
+			"{\"name\":\"amount\",\"type\":\"uint256\"}],\"name\":\"deposit\",\"outputs\":[]," +
+			"\"payable\":true,\"stateMutability\":\"payable\",\"type\":\"function\"}";
```
Comment (qnswirlds) [inline test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/meta/ContractResources.java:325]: I believe these 2 are duplicates of `DEPOSIT_ABI` and `GET_BALANCE_ABI`. Please double check and remove them if identical.
Also, some `nits`:
- Typo `JURIDICTION` -> `JURISDICTION`
- The ABI string format is preferable to split the `input` and `output` into two separate lines, (the `constant` thing is on the line before the input, and the `payable` thing is on the line after the output. If those two lines are too long, split more at the end of each parameter in the input/output.

Comment (anighanta) [reply test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/meta/ContractResources.java]: Thanks @QianSwirlds. Fixed them

**Review by qnswirlds — APPROVED** [review]
LGTM! Thanks for the changes!

**Pull request merged**